community/cloud-foundation/templates/backend_service/backend_service.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" This template creates a backend service. """
+
+REGIONAL_GLOBAL_TYPE_NAMES = {
+    True: 'compute.v1.regionBackendService',
+    False: 'compute.v1.backendService'
+}
+
+
+def set_optional_property(destination, source, prop_name):
+    """ Copies the property value if present. """
+
+    if prop_name in source:
+        destination[prop_name] = source[prop_name]
+
+
+def get_backend_service_outputs(res_name, backend_name, region):
+    """ Creates outputs for the backend service. """
+
+    outputs = [
+        {
+            'name': 'name',
+            'value': backend_name
+        },
+        {
+            'name': 'selfLink',
+            'value': '$(ref.{}.selfLink)'.format(res_name)
+        }
+    ]
+
+    if region:
+        outputs.append({'name': 'region', 'value': region})
+
+    return outputs
+
+
+def generate_config(context):
+    """ Entry point for the deployment resources. """
+
+    properties = context.properties
+    res_name = context.env['name']
+    name = properties.get('name', res_name)
+    is_regional = 'region' in properties
+    region = properties.get('region')
```

Comment:
Please use properties['region'] instead of .get everywhere. .get('name', res_name) can be an exception because of the default value.

Reply:
@ocsig Sorry, I'm not sure I completely understand. `region` and `name` are optional by design, they both intentionally don't have default schema value and therefore, in order to avoid `KeyError`, retrieved via `.get`.
`region` is later used for branching and conditional assignment, so its potential `None` value is perfectly valid.

Reply:
Ok, it was my bad.